libtsuba/src/TxnContext.cpp

```diff
@@ -0,0 +1,25 @@
+#include "katana/TxnContext.h"
+
+#include "GlobalState.h"
+#include "katana/file.h"
+
+katana::Result<void>
+katana::TxnContext::Commit() const {
+  if (manifest_file_.empty()) {
+    return katana::ResultSuccess();
+  }
+
+  katana::CommBackend* comm = katana::Comm();
+  comm->Barrier();
+
```

Comment:
Why do we need this Barrier?

Reply:
Please avoid barriers like the plague in the code ;)

It's better to either use `DistTerminationDetection` or `CombineResultAndReset()`

Reply:
@roshandathathri looks like both of the functions are in enterprise repo and not exposed to the open repo... is there any other substitutes? I right now get a segfault in `katana::Comm()` while calling this `TxnContext.Commit()` in the `~TxnContext()` when the scope of the `TxnContext` is in `main`. If there's no better way, I'll just modify the tests to explicitly call a `TxnContext.Commit()` before the program exits.

Reply:
The barrier code was copied from the old `CommitRDG` in `libtsuba/src/RDG.cpp`, for making sure the writer of OneHostOnly isn't writing a previous version of the file.

Reply:
Y'know what?  There is a barrier inside `OneHostOnly`.  I bet it wasn't there when this Barrier was added.  @tylershunt might remember off hand.  But I think we can get rid of this `comm->Barrier()`;

The issue still remains that you don't want to call commit after the end of main.

Reply:
I see... Looking like `OneHostOnly` already calls Comm()->Barrier() by itself. Thanks!